src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +178,32 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let mut has_target = false;
+        let target_str = target.to_string();
+
+        let supported_target_list =
+            output(Command::new(&build.config.initial_rustc).args(["--print", "target-list"]));
+
+        has_target |= supported_target_list.contains(&target_str);
+
+        // If not, check for a valid file location that may have been specified
+        // by the user for the custom target.
+        if let Some(custom_target_path) = env::var_os("RUST_TARGET_PATH") {
+            let target_os_str = OsStr::new(&target_str);
+            let walker = WalkDir::new(custom_target_path).into_iter();
+            for entry in walker.filter_map(|e| e.ok()) {
+                has_target |= entry.file_name() == target_os_str;
```

Comment:
Target files end with `.json`, so for `--target x86_64-some-custom-target` we should look for `x86_64-some-custom-target.json` file.